sources/packages/web/src/views/student/StudentApplicationSummary.vue

```diff
@@ -38,14 +57,86 @@ export default {
     StudentApplications,
     CheckValidSINBanner,
     HeaderNavigator,
+    ConfirmEditApplication,
+    CancelApplication,
   },
   setup() {
     const hasRestriction = ref(false);
     const restrictionMessage = ref("");
+    const programYear = ref({} as ProgramYearOfApplicationDto);
+    const router = useRouter();
+    const toast = useToastMessage();
+    const TOAST_ERROR_DISPLAY_TIME = 15000;
+    const editApplicationModal = ref({} as ModalDialog<boolean>);
+    const showModal = ref(false);
+    const selectedApplicationId = ref(0);
+    const reloadData = ref(false);
+
+    const openConfirmCancel = (id: number) => {
+      showModal.value = true;
+      selectedApplicationId.value = id;
+    };
+
+    const showHideCancelApplication = () => {
+      showModal.value = !showModal.value;
+    };
+
+    const getProgramYear = async (applicationId: number) => {
+      programYear.value = await ApplicationService.shared.getProgramYearOfApplication(
+        applicationId,
+      );
+    };
+
+    const editApplication = async (applicationId: number) => {
+      try {
+        await getProgramYear(applicationId);
+        router.push({
+          name: StudentRoutesConst.DYNAMIC_FINANCIAL_APP_FORM,
+          params: {
+            selectedForm: programYear.value.formName,
+            programYearId: programYear.value.programYearId,
+            id: applicationId,
+          },
+        });
+      } catch (error) {
+        toast.error(
+          "Program Year not active",
```

Comment:
If there is an error on getProgramYear() API call, can't there be any other reason except program year not being active.

Reply:
I believe, this API is looking for an active program year while editing, 
![image](https://user-images.githubusercontent.com/77353155/155208096-cdff2a2f-72be-4b13-8439-572eebe64ad3.png)


these vue changes are part of refactoring, I move this function  from the shared component to the student's parent component calling the shared component

Reply:
As discussed we can note this down and add to technical debt and dev discussion point at our next scheduled sync up meeting.

Reply:
added this note in https://app.zenhub.com/workspaces/student-information-management-system-5fce9df5aa1b45000e937014/issues/bcgov/sims/695